HABITAT_2_TESTING_GUIDE.md

```diff
@@ -0,0 +1,410 @@
+
+**Linux:**
+```bash
+curl https://raw.githubusercontent.com/habitat-sh/habitat/main/components/hab/migrate.sh | \
+  sudo -E bash -s -- --channel hab-2-rc1
+```
+
+**Windows:**
+```powershell
+iex "& { $(irm https://raw.githubusercontent.com/habitat-sh/habitat/main/components/hab/migrate.ps1) } -Channel hab-2-rc1"
+```
+
+### Verification
+
+```bash
+hab --version
+# Expected: habitat 2.x.x
+
+sudo hab svc status
+# Verify all services are running with Habitat 2.0
+```
+
+---
+
+**Note:** The doc has not been tested from this point onwards.
+
+## Scenario 5: Workstation Setup (Including ARM64)
+
+**Objective:** Install habitat on a workstation (Linux ARM, x86_64, or Windows) and run Habitat applications
+
+### Prerequisites
+
+- [ ] Workstation system (ARM64, x86_64 Linux, or Windows)
+- [ ] Network access to on-prem builder
+- [ ] Authentication token for on-prem builder
+
+### Step 1: Install Habitat on Workstation
+
+Follow the installation steps from **Scenario 3** based on your workstation architecture.
+
+### Step 2: Configure Workstation
+
+📖 **Reference:** [Workstation setup guide](./docs-chef-io/content/habitat/on_prem_builder/install/workstation.md)
+
+```bash
+# Point to your on-prem builder
+export HAB_BLDR_URL=https://your-builder.example.com/bldr/v1/
+
+# Set authentication token
+export HAB_AUTH_TOKEN=<ON_PREM_BUILDER_TOKEN>
+
+# For self-signed certificates (if applicable)
+export SSL_CERT_FILE=/path/to/ssl-certificate.crt
+```
+
+### Step 3: Test Package Installation
+
+```bash
+# Test installing a package from your on-prem builder
+hab pkg install core/curl
+
+# Verify installation
+hab pkg path core/curl
+```
+
+--
+
+## Scenario 6: Custom Package Development
+
+**Requirement**: "download, install, build a custom hab package/service using base channel deps"
+
+This scenario tests building custom packages with Habitat 2.0 dependencies:
+
+### On the On Prem Builder Instance
+
+Run the following command on the system with On Prem Builder Setup:
+
+```bash
+# 1. Create a test plan that uses base-2025 dependencies
+mkdir test-package
+cd test-package
+
+# 2. Create plan.sh with base-2025 deps
+cat > plan.sh << 'EOF'
+pkg_name=test-package
+pkg_origin=test
+pkg_version="1.0.0"
+pkg_deps=(core/glibc)
+pkg_build_deps=(core/gcc)
+
+do_build() {
+  echo "Building with Habitat 2.0 base deps"
+}
+
+do_install() {
+  echo "Installing test package"
+}
+EOF
+
+# 4. One time command to generate the public key at ~/.hab/cache/keys
+sudo hab origin key generate test
+
+# 5. Set env
+export HAB_ORIGIN=test
+
+#Required for private packages
+export HAB_AUTH_TOKEN=<your private bldr token> 
+
+
+# 3. Build using Habitat 2.0 with base-2025 channel
+sudo-E  hab pkg build . --refresh-channel base-2025
+
+#4 Upload to on prem bldr
+sudo -E hab pkg upload <PKG_IDENT>
+```
+
+### On the workstation
+
+Run the followong command on the Workstation:
+
+```bash
+# Download the package inside the workstation
+sudo hab pkg download <PKG_IDENT>
```

Comment:
no need for a `hab pkg download` because `hab pkg install` will download the package.

Reply:
Removed hab pkg download in commit [a117cfd](https://github.com/habitat-sh/on-prem-builder/pull/292/commits/a117cfdc73d5f87c9c1654ccbd1632840d5ec41f)